sklearn/decomposition/_nmf.py

```diff
@@ -1304,24 +1246,97 @@ def fit_transform(self, X, y=None, W=None, H=None):
         W : ndarray of shape (n_samples, n_components)
             Transformed data.
         """
-        X = self._validate_data(X, accept_sparse=('csr', 'csc'),
-                                dtype=[np.float64, np.float32])
```

Comment:
This was the correct way to validate X

Reply:
Yes, sorry I wasn't meant to delete those lines.